Makefile

```diff
@@ -4,30 +4,81 @@ RISC0_DIR := $(ROOT_DIR)/risc0_benchmarks
 SP1_DIR := $(ROOT_DIR)/sp1_benchmarks
 RESULTS_DIR := $(ROOT_DIR)/results
 
+
+
+# Helper target to cleanup background processes
+.PHONY: cleanup
+cleanup:
+	@rm -f $(RESULTS_DIR)/.top.pid
+	@rm -f $(RESULTS_DIR)/risc0_cpu_usage.log
+	@rm -f $(RESULTS_DIR)/risc0_memory_leak.log
+	@rm -f $(RESULTS_DIR)/risc0_rust_bench.log
+	@rm -f $(RESULTS_DIR)/.sp1top.pid
+	@rm -f $(RESULTS_DIR)/sp1_cpu_usage.log
+	@rm -f $(RESULTS_DIR)/sp1_memory_leak.log
+#	@rm -f $(RESULTS_DIR)/sp1_rust_bench.log
+	@-pkill -f "top -b -d 1" || true
+
+
+
 # Target for RISC Zero project
 .PHONY: risc0
-risc0:
-	@if [ -z "$(PROJECT)" ]; then \
-	    echo "Error: Please specify a PROJECT variable. Example: make risc0 PROJECT=test_project"; \
-	    exit 1; \
-	fi
-	@echo "Running RISC Zero benchmarks for project: $(PROJECT)"
+risc0: cleanup
+	@if [ -z "$(TEST_NAME)" ]; then \
+        echo "Error: Please specify a TEST_NAME variable. Example: make risc0 PROJECT=test_project TEST_NAME=test_name"; \
+        exit 1; \
+    fi
+	@echo "Running RISC Zero benchmarks for: $(TEST_NAME)"
 	@mkdir -p $(RESULTS_DIR)
-	@cd $(RISC0_DIR)/$(PROJECT)/methods && cargo build --release 
-	@cd $(RISC0_DIR)/$(PROJECT)/host && cargo build --release
-	@cd $(RISC0_DIR)/$(PROJECT)/host && cargo run --release > $(RESULTS_DIR)/risc0_$(PROJECT)_results.txt
-	@echo "RISC Zero benchmarks completed! Results saved to $(RESULTS_DIR)/risc0_$(PROJECT)_results.txt"
+	@cd $(RISC0_DIR)/test_project/methods && cargo build --release
+	@cd $(RISC0_DIR)/test_project/host && cargo build --release
+	@top -b -d 1 -n 0 | head -n 5 > $(RESULTS_DIR)/risc0_cpu_usage.log & echo $$! > $(RESULTS_DIR)/.top.pid
```

Comment:
bu kisim napiyo? -n 0 argumani bende hata veriyo

Reply:
kanka benchmark bittikten sonra top'u öldürmek için lazımdı da artık gerek yok sildim o kısmı